File: client/cocos/assets/scripts/frame/mvc/BaseModel.ts
export class BaseModel {
  private changeCallbacks: Map<string, ((value: any) => void)[]> = new Map();

  // Register change listener for any field
  onChange(field: string, callback: (value: any) => void) {
    if (!this.changeCallbacks.has(field)) {
      this.changeCallbacks.set(field, []);
    }
    this.changeCallbacks.get(field)!.push(callback);
    return () => {
      const callbacks = this.changeCallbacks.get(field);
      if (callbacks) {
        const index = callbacks.indexOf(callback);
        if (index > -1) {
          callbacks.splice(index, 1);
        }
      }
    };
  }

  // Notify change for any field
  protected notifyChange(field: string, value: any) {
    const callbacks = this.changeCallbacks.get(field);
    if (callbacks) {
      callbacks.forEach((callback) => callback(value));
    }
  }
}

File: client/unity/Assets/Scripts/frame/Mvc/BaseModel.cs
using System;
using System.Collections.Generic;

namespace Nova.Mvc
{
    public class BaseModel
    {
        // Register change listener for any field
        private Dictionary<string, LinkedList<Action<object>>> _changeCallbacks = new Dictionary<string, LinkedList<Action<object>>>();
        
        public void Register(string fieldName, Action<object> callback)
        {
            if (!_changeCallbacks.ContainsKey(fieldName))
            {
                _changeCallbacks[fieldName] = new LinkedList<Action<object>>();
            }
            _changeCallbacks[fieldName].AddLast(callback);
        }
        
        public void Notify(string fieldName, object value)
        {
            if (_changeCallbacks.ContainsKey(fieldName))
            {
                foreach (var callback in _changeCallbacks[fieldName])
                {
                    callback(value);
                }
            }
        }
    }
}